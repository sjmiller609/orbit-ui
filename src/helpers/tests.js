import { BrowserRouter } from 'react-router-dom'
import { shape } from 'prop-types'
import { mount } from 'enzyme'

// Instantiate router context
const router = {
  history: new BrowserRouter().history,
  route: {
    location: {},
    match: {},
  },
}

const createContext = () => ({
  context: { router },
  childContextTypes: { router: shape({}) },
})

// wrap with router context - to mount RR links correctly
export function mountWrap(node) {
  return mount(node, createContext())
}
