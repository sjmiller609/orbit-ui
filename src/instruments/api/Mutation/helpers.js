const has = (str, find) => ~str.toLowerCase().indexOf(find.toLowerCase())

export const formErrors = ({ error, errors }) => {
  if (!error) return

  let err = JSON.stringify(error).toLowerCase()
  return errors.find(e => {
    if (e.key && has(err, e.key)) return e
    else if (e.allKeys && e.allKeys.every(k => has(err, k))) return e
    else if (e.someKeys && e.someKeys.some(k => has(err, k))) return e
  })
}
