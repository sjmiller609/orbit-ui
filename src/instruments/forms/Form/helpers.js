// faltten and stringify nested keys
export const unpack = (obj, ns, hash) => {
  if (!obj) return
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keypath = ns ? `${ns}.${key}` : key
    if (value && typeof value === 'object') {
      return unpack(value, keypath, acc)
    }
    acc[keypath] = value
    return acc
  }, hash || {})
}

const roll = (keys, obj, value) => {
  const key = keys[0]
  // case 1 - not obj
  if (keys.length === 1) {
    obj[key] = value
    return obj
  }
  // case 2 - doesn't exist
  if (!obj[key]) {
    obj[key] = isNaN(keys[1]) ? {} : []
  }
  // case 3 - exists
  obj[key] = roll(
    keys.slice(1),
    Array.isArray(obj[key]) ? [...obj[key]] : { ...obj[key] },
    value
  )

  return obj
}

// convert back into obj
export const pack = (obj, type = {}) => {
  return Object.entries(obj).reduce((acc, [key, value]) => {
    const keys = key.split('.')

    return roll(keys, acc, value)
  }, type)
}

export const packChild = ({ name, obj }) => {
  const obj2 = {}
  let isArray
  Object.entries(obj).forEach(k => {
    if (name.length < k[0].length && k[0].indexOf(name) === 0) {
      const n = k[0].slice(name.length + 1)
      // must be true for all matching
      isArray = !isNaN(n.slice(0, n.indexOf('.')))
      obj2[n] = k[1]
    }
  })
  if (Object.keys(obj2).length) return pack(obj2, isArray ? [] : {})
  return null
}

export const removeChild = ({ name, obj = {} }) => {
  const obj2 = {}
  Object.entries(obj).forEach(k => {
    if (name.length >= k[0].length || k[0].indexOf(name) !== 0) {
      obj2[k[0]] = k[1]
    }
  })
  return obj2
}
