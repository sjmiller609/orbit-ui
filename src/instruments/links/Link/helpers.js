// check if url is external
export const externalUrl = url => {
  if (typeof url === 'object') return false
  const path = /^https?:\/\/|^\/\//i
  if (path.test(url) && !~url.indexOf(window.location.origin)) return true
  return
}
