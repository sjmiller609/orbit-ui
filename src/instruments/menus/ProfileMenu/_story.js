import React from 'react'
import { storiesOf } from '@storybook/react'
import { ProfileMenu } from 'instruments'

storiesOf('Instruments|Menus', module).add('ProfileMenu', () => (
  <ProfileMenu name="John Doe" avatar="null" platform={true} />
))
