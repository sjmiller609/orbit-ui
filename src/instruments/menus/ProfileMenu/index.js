'use strict'
import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import { Avatar, MenuList, Item, Dropdown } from 'instruments'
import s from './styles.scss'

const ProfileMenu = ({ name, avatar, className }) => {
  return (
    <Dropdown
      className={classnames(s.menu, className)}
      right
      selector={<Avatar url={avatar} title={name} className={s.profile} />}>
      <MenuList label={name}>
        <Item to="/profile">Personal Settings</Item>
        <Item to="/logout">Logout</Item>
      </MenuList>
    </Dropdown>
  )
}

ProfileMenu.propTypes = {
  name: PropTypes.string,
  avatar: PropTypes.string,
  className: PropTypes.string,
}

export default ProfileMenu
