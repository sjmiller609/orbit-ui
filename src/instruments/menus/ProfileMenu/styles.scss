@import 'src/styles/mixins.scss';

$size: 3.5rem;

.icon {
  height: $size;
  width: $size;
  border-radius: 100rem;
  box-shadow: 0 0 0 2px $lightGray;
  background: $gray;
}

.profile {
  box-shadow: 0 0 0 2px $lightGray;
  background: $gray;
  img {
    height: 100%;
    &:hover {
      transform: rotate(12deg);
      transition: 250ms ease-in;
    }
  }
}

.title {
  display: block;
  margin: 1.2rem 0;
}
