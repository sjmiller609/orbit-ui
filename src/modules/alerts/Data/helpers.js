import { formErrors } from 'instruments'

const errors = [
  {
    key: 'permissions',
    name: 'properties.alert_emails',
    error: 'Contact your workspace admin to upgrade your permissions.',
  },
]

export const handleError = error => formErrors({ error, errors })

export const trimError = error => error.split(':')[1].trim()
