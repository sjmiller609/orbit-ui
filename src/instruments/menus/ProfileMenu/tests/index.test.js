import React from 'react'
import { mountWrap } from 'helpers/tests'
import ProfileMenu from '../index'

describe('ProfileMenu', () => {
  it('should be defined', () => {
    expect(ProfileMenu).toBeDefined()
  })

  it('should render correctly', () => {
    const wrapper = mountWrap(<ProfileMenu />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render admin link', () => {
    const wrapper = mountWrap(<ProfileMenu platform={true} />)
    expect(wrapper).toMatchSnapshot()
  })
})
