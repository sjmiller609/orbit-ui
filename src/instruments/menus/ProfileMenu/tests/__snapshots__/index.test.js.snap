// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMenu should render admin link 1`] = `
<ProfileMenu
  platform={true}
>
  <Dropdown
    className="menu"
    right={true}
    selector={
      <Avatar
        className="profile"
        title={undefined}
        url={undefined}
      />
    }
  >
    <div
      className="dropdown right out menu"
    >
      <Backdrop
        close={[Function]}
        show={false}
      />
      <Link
        className="selector"
        onClick={[Function]}
      >
        <A
          className="link selector"
          onClick={[Function]}
        >
          <a
            className="link selector"
            onClick={[Function]}
          >
            <Avatar
              className="profile"
            >
              <Box
                className="avatar profile"
              >
                <Flex
                  className="avatar profile"
                  flow="box"
                >
                  <div
                    className="flex box undefined avatar profile"
                  >
                    <Icon
                      icon="astro_helmet"
                    >
                      <Imported>
                        <ReactImportedComponent
                          loadable={
                            Object {
                              "done": true,
                              "importFunction": [Function],
                              "load": [Function],
                              "mark": Array [],
                              "ok": true,
                              "payload": Object {
                                "process": [Function],
                              },
                              "promise": Promise {},
                              "reset": [Function],
                            }
                          }
                          render={[Function]}
                        >
                          <UnconnectedReactImportedComponent
                            async={false}
                            exportPicker={[Function]}
                            loadable={
                              Object {
                                "done": true,
                                "importFunction": [Function],
                                "load": [Function],
                                "mark": Array [],
                                "ok": true,
                                "payload": Object {
                                  "process": [Function],
                                },
                                "promise": Promise {},
                                "reset": [Function],
                              }
                            }
                            render={[Function]}
                            streamId={0}
                          >
                            <img
                              src={
                                Object {
                                  "process": [Function],
                                }
                              }
                            />
                          </UnconnectedReactImportedComponent>
                        </ReactImportedComponent>
                      </Imported>
                    </Icon>
                  </div>
                </Flex>
              </Box>
            </Avatar>
          </a>
        </A>
      </Link>
      <div
        className="menuWrapper"
        onClick={[Function]}
      >
        <Card
          className="menu"
        >
          <Box
            className="card menu"
            justify="flex-start"
          >
            <Flex
              className="card menu"
              flow="box"
              justify="flex-start"
            >
              <div
                className="flex box flex-start card menu"
              >
                <main
                  className="full"
                >
                  <MenuList>
                    <ul
                      className="list"
                    >
                      <Item
                        exact={true}
                        to="/profile"
                      >
                        <Link
                          activeClassName="active"
                          className="item"
                          exact={true}
                          title="Personal Settings"
                          to="/profile"
                        >
                          <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="link item"
                            exact={true}
                            title="Personal Settings"
                            to="/profile"
                          >
                            <Route
                              exact={true}
                              path="\\\\/profile"
                            >
                              <Link
                                aria-current={null}
                                className="link item"
                                replace={false}
                                title="Personal Settings"
                                to="/profile"
                              >
                                <a
                                  aria-current={null}
                                  className="link item"
                                  href="/profile"
                                  onClick={[Function]}
                                  title="Personal Settings"
                                >
                                  Personal Settings
                                </a>
                              </Link>
                            </Route>
                          </NavLink>
                        </Link>
                      </Item>
                      <Item
                        exact={true}
                        to="/admin"
                      >
                        <Link
                          activeClassName="active"
                          className="item"
                          exact={true}
                          title="Admin Settings"
                          to="/admin"
                        >
                          <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="link item"
                            exact={true}
                            title="Admin Settings"
                            to="/admin"
                          >
                            <Route
                              exact={true}
                              path="\\\\/admin"
                            >
                              <Link
                                aria-current={null}
                                className="link item"
                                replace={false}
                                title="Admin Settings"
                                to="/admin"
                              >
                                <a
                                  aria-current={null}
                                  className="link item"
                                  href="/admin"
                                  onClick={[Function]}
                                  title="Admin Settings"
                                >
                                  Admin Settings
                                </a>
                              </Link>
                            </Route>
                          </NavLink>
                        </Link>
                      </Item>
                      <Item
                        exact={true}
                        to="/logout"
                      >
                        <Link
                          activeClassName="active"
                          className="item"
                          exact={true}
                          title="Logout"
                          to="/logout"
                        >
                          <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="link item"
                            exact={true}
                            title="Logout"
                            to="/logout"
                          >
                            <Route
                              exact={true}
                              path="\\\\/logout"
                            >
                              <Link
                                aria-current={null}
                                className="link item"
                                replace={false}
                                title="Logout"
                                to="/logout"
                              >
                                <a
                                  aria-current={null}
                                  className="link item"
                                  href="/logout"
                                  onClick={[Function]}
                                  title="Logout"
                                >
                                  Logout
                                </a>
                              </Link>
                            </Route>
                          </NavLink>
                        </Link>
                      </Item>
                    </ul>
                  </MenuList>
                </main>
              </div>
            </Flex>
          </Box>
        </Card>
      </div>
    </div>
  </Dropdown>
</ProfileMenu>
`;

exports[`ProfileMenu should render correctly 1`] = `
<ProfileMenu>
  <Dropdown
    className="menu"
    right={true}
    selector={
      <Avatar
        className="profile"
        title={undefined}
        url={undefined}
      />
    }
  >
    <div
      className="dropdown right out menu"
    >
      <Backdrop
        close={[Function]}
        show={false}
      />
      <Link
        className="selector"
        onClick={[Function]}
      >
        <A
          className="link selector"
          onClick={[Function]}
        >
          <a
            className="link selector"
            onClick={[Function]}
          >
            <Avatar
              className="profile"
            >
              <Box
                className="avatar profile"
              >
                <Flex
                  className="avatar profile"
                  flow="box"
                >
                  <div
                    className="flex box undefined avatar profile"
                  >
                    <Icon
                      icon="astro_helmet"
                    >
                      <Imported>
                        <ReactImportedComponent
                          loadable={
                            Object {
                              "done": true,
                              "importFunction": [Function],
                              "load": [Function],
                              "mark": Array [],
                              "ok": true,
                              "payload": Object {
                                "process": [Function],
                              },
                              "promise": Promise {},
                              "reset": [Function],
                            }
                          }
                          render={[Function]}
                        >
                          <UnconnectedReactImportedComponent
                            async={false}
                            exportPicker={[Function]}
                            loadable={
                              Object {
                                "done": true,
                                "importFunction": [Function],
                                "load": [Function],
                                "mark": Array [],
                                "ok": true,
                                "payload": Object {
                                  "process": [Function],
                                },
                                "promise": Promise {},
                                "reset": [Function],
                              }
                            }
                            render={[Function]}
                            streamId={0}
                          >
                            <img
                              src={
                                Object {
                                  "process": [Function],
                                }
                              }
                            />
                          </UnconnectedReactImportedComponent>
                        </ReactImportedComponent>
                      </Imported>
                    </Icon>
                  </div>
                </Flex>
              </Box>
            </Avatar>
          </a>
        </A>
      </Link>
      <div
        className="menuWrapper"
        onClick={[Function]}
      >
        <Card
          className="menu"
        >
          <Box
            className="card menu"
            justify="flex-start"
          >
            <Flex
              className="card menu"
              flow="box"
              justify="flex-start"
            >
              <div
                className="flex box flex-start card menu"
              >
                <main
                  className="full"
                >
                  <MenuList>
                    <ul
                      className="list"
                    >
                      <Item
                        exact={true}
                        to="/profile"
                      >
                        <Link
                          activeClassName="active"
                          className="item"
                          exact={true}
                          title="Personal Settings"
                          to="/profile"
                        >
                          <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="link item"
                            exact={true}
                            title="Personal Settings"
                            to="/profile"
                          >
                            <Route
                              exact={true}
                              path="\\\\/profile"
                            >
                              <Link
                                aria-current={null}
                                className="link item"
                                replace={false}
                                title="Personal Settings"
                                to="/profile"
                              >
                                <a
                                  aria-current={null}
                                  className="link item"
                                  href="/profile"
                                  onClick={[Function]}
                                  title="Personal Settings"
                                >
                                  Personal Settings
                                </a>
                              </Link>
                            </Route>
                          </NavLink>
                        </Link>
                      </Item>
                      <Item
                        exact={true}
                        to="/logout"
                      >
                        <Link
                          activeClassName="active"
                          className="item"
                          exact={true}
                          title="Logout"
                          to="/logout"
                        >
                          <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="link item"
                            exact={true}
                            title="Logout"
                            to="/logout"
                          >
                            <Route
                              exact={true}
                              path="\\\\/logout"
                            >
                              <Link
                                aria-current={null}
                                className="link item"
                                replace={false}
                                title="Logout"
                                to="/logout"
                              >
                                <a
                                  aria-current={null}
                                  className="link item"
                                  href="/logout"
                                  onClick={[Function]}
                                  title="Logout"
                                >
                                  Logout
                                </a>
                              </Link>
                            </Route>
                          </NavLink>
                        </Link>
                      </Item>
                    </ul>
                  </MenuList>
                </main>
              </div>
            </Flex>
          </Box>
        </Card>
      </div>
    </div>
  </Dropdown>
</ProfileMenu>
`;
